include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
+
+#include <stdint.h>
+
+#undef min
+#undef max
+#undef DeleteFile
+#undef GetCurrentTime
+
+#ifndef strcasecmp
+#define strcasecmp _stricmp
+#endif
+
+#ifndef snprintf
+#define snprintf _snprintf
+#endif
+
+#ifndef ROCKSDB_PRIszt
+#define ROCKSDB_PRIszt "Iu"
+#endif
```

Comment:
The recurring theme both in db.h and env.h are macros that redefine DeleteFile, GetCurrentTime.

Although there are still #undefs remaining from previous porting efforts within the headers which we should remove because having it in one file is much cleaner.

Perf_context.h requires __thread (tls) keyword.

Snprintf is also a function that we hit in the past and sure enough will hit in the future.

Thanks,
Dmitri

From: Siying Dong [mailto:notifications@github.com]
Sent: Monday, July 13, 2015 6:24 PM
To: facebook/rocksdb
Cc: Dmitri Smirnov
Subject: Re: [rocksdb] Ensure Windows build w/o port/port.h in public headers (#652)


In include/rocksdb/public_port.h<https://github.com/facebook/rocksdb/pull/652#discussion_r34530079>:

> +#undef min

> +#undef max

> +#undef DeleteFile

> +#undef GetCurrentTime

> +

> +#ifndef strcasecmp

> +#define strcasecmp _stricmp

> +#endif

> +

> +#ifndef snprintf

> +#define snprintf _snprintf

> +#endif

> +

> +#ifndef ROCKSDB_PRIszt

> +#define ROCKSDB_PRIszt "Iu"

> +#endif

Can you give examples?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/rocksdb/pull/652/files#r34530079>.